File: CanliYayinSwift/CanliYayinSwift/model/ChatMessageModel.swift
//
//  ChatMessageModel.swift
//  CanliYayinSwift
//
//  Created by Alperen Saraç on 13.05.2026.
//

import Foundation

struct ChatMessageModel {
    let roomId: String
    let username: String
    let message: String
    let createdAt: String

    init(json: [String: Any]) {
        self.roomId = json["room_id"] as? String ?? ""
        self.username = json["username"] as? String ?? ""
        self.message = json["message"] as? String ?? ""
        self.createdAt = json["created_at"] as? String ?? ""
    }
}

File: CanliYayinJava/app/src/main/java/com/example/canliyayinjava/model/ChatMessageModel.java
package com.example.canliyayinjava.model;

public class ChatMessageModel {

    private String username;
    private String message;
    private String createdAt;

    public ChatMessageModel(String username, String message, String createdAt) {
        this.username = username;
        this.message = message;
        this.createdAt = createdAt;
    }

    public String getUsername() {
        return username;
    }

    public String getMessage() {
        return message;
    }

    public String getCreatedAt() {
        return createdAt;
    }
}
